app/controllers/fhir/r4/api_controller.rb

```diff
@@ -426,6 +426,123 @@ def all
+      status_not_acceptable_with_custom_errors(["'Accept' header must have a value of 'application/fhir+json'," \
+        " or the '_format' parameter must be one of 'json', 'application/json' or 'application/fhir+json'"]) && return
+    end
+
+    unless prefer_header?
+      status_unprocessable_entity_with_custom_errors(["'Prefer' header must have value of 'respond-async'"],
+                                                     '') && return
+    end
+
+    unless @m2m_workflow
+      status_unauthorized_with_custom_errors(['Bulk export requires a client application registered for the Backend Services Workflow']) && return
+    end
+
+    if params.key?(:_type) || params.key?(:_outputFormat)
+      status_unprocessable_entity_with_custom_errors(['The _type and _outputFormat parameters are unsupported'], '') && return
+    end
+
+    begin
+      since = params.permit(:_since)[:_since]
+      since = DateTime.strptime(since, '%Y-%m-%dT%H:%M:%S%z') unless since.blank?
+    rescue Date::Error
+      status_unprocessable_entity_with_custom_errors(['Invalid Date in _since parameter. Please use the FHIR instant datatype'], '') && return
+    end
+
+    client_app = current_client_application
+    if client_app.exported_recently?
+      status_too_many_requests_with_custom_errors(['Client already initiated an export of this type in the last 15 minutes. Please try again later']) && return
+    end
+
+    # Create the download to uniquely id this request, and queue the ExportFhirJob
+    download = ApiDownload.create(application_id: client_app.id, url: request.url)
+    export_job = ExportFhirJob.perform_later(client_app, download, { since: since })
+    # Add the job_id of the job to the download for reference in status requests
+    download.update(job_id: export_job.provider_job_id)
+
+    respond_to do |format|
+      format.any { head(:accepted, content_location: "#{root_url}fhir/r4/ExportStatus/#{download.id}") }
+    end
+  end
+
+  # Give an update on the status of a bulk FHIR data request
+  #
+  # GET /fhir/r4/ExportStatus/[:id]
+  def export_status
+    id = params.require(:id)
+    download = ApiDownload.find_by_id(id)
+    status = Sidekiq::Status.status(download&.job_id)
+    case status
+    when :complete
+      begin
+        transaction_time = DateTime.parse(Sidekiq::Status.get(download.job_id, :transaction_time))
+      rescue Date::Error
+        status_server_error(['Export failed']) && return
+      end
+
+      response_json = {
+        transactionTime: transaction_time.utc.strftime('%FT%T%:z'),
+        request: download.url,
+        requiresAccessToken: true,
+        output: download.files.blobs.pluck(:filename).map do |file|
+          resource_type = file.split('.').first
+          {
+            type: resource_type,
+            url: "#{root_url}fhir/r4/ExportFiles/#{id}/#{resource_type}"
+          }
+        end
+      }
+      response.headers['Expires'] = Chronic.parse(ADMIN_OPTIONS['weekly_purge_date']).httpdate
```

Comment:
Considering the verbage in the spec for the Expires header
>indicating when the files listed will no longer be available for access

I'd say this usage is a little misleading if we're purging old exports after new ones are processed. Since this is just a "MAY" in the spec, how would you feel about removing it?

Reply:
Addressed in https://github.com/SaraAlert/SaraAlert/pull/987/commits/5e43859283d46a982023b4d0a5965c6e314d7174.